src/services/expenses.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+let expenses = [];
+
+const init = () => {
+  expenses = [];
+};
+
+const getAll = (query) => {
+  const { userId, categories, from, to } = query;
+
+  return expenses.filter(expense => {
+    const userIdMatch = userId
+      ? +expense.userId === +userId
+      : true;
+
+    const categoriesMatch = categories
+      ? categories.includes(expense.category)
+      : true;
+
+    const datesMatch = (from && to)
+      ? expense.spentAt >= from && expense.spentAt <= to
+      : true;
+
+    return userIdMatch && categoriesMatch && datesMatch;
```

Comment:
ok, but we can use `if` to return false on the first `false` result and avoid next calculations